includes/abilities/class-wp-core-abilities.php

```diff
@@ -0,0 +1,286 @@
+<?php
+/**
+ * Core Abilities registration.
+ *
+ * @package WordPress
+ * @subpackage Abilities_API
+ * @since 0.3.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Registers the default core abilities that ship with the Abilities API.
+ *
+ * @since 0.3.0
+ */
+// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedClassFound -- Core class intended for WordPress core.
+class WP_Core_Abilities {
+	/**
+	 * Category slugs for core abilities.
+	 *
+	 * @since 0.3.0
+	 */
+	public const CATEGORY_SITE = 'site';
+	public const CATEGORY_USER = 'user';
+	/**
+	 * Registers the core abilities categories.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return void
+	 */
+	public static function register_category(): void {
+		// Site-related capabilities
+			wp_register_ability_category(
+				self::CATEGORY_SITE,
+				array(
+					'label'       => __( 'Site' ),
+					'description' => __( 'Abilities that retrieve or modify site information and settings.' ),
+				)
+			);
+
+		// User-related capabilities
+			wp_register_ability_category(
+				self::CATEGORY_USER,
+				array(
+					'label'       => __( 'User' ),
+					'description' => __( 'Abilities that retrieve or modify user information and settings.' ),
+				)
+			);
+	}
+
+	/**
+	 * Registers the default core abilities.
+	 *
+ * @since 0.3.0
+	 *
+	 * @return void
+	 */
+	public static function register(): void {
+		self::register_get_site_info();
+		self::register_get_current_user_info();
+		self::register_get_environment_info();
+	}
+
+	/**
+	 * Registers the `wp/get-site-info` ability.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return void
+	 */
+	protected static function register_get_site_info(): void {
+		$fields = array(
+			'name',
+			'description',
+			'url',
+			'wpurl',
+			'admin_email',
+			'charset',
+			'language',
+			'version',
+		);
+
+		wp_register_ability(
+			'wp/get-site-info',
+			array(
+				'label'               => __( 'Get Site Information' ),
+				'description'         => __( 'Returns a single site information field configured in WordPress (e.g., site name, URL, version) for display or diagnostics.' ),
+				'category'            => self::CATEGORY_SITE,
+				'input_schema'        => array(
+					'type'                 => 'object',
+					'properties'           => array(
+						'field' => array(
+							'type'        => 'string',
+							'enum'        => $fields,
+							'description' => __( 'The site information field to retrieve.' ),
+						),
+					),
```

Comment:
I'm going to be very curious to see how well the AI uses this. If each field was a distinct property with its own description, that would probably work the best, but I get why we're doing this. If it doesn't work well, then we could have each field as a boolean, where the AI determines which fields it wants and which it doesn't, returning multiple at once.